File: final_prog/backtracking/2_1DQueenPermutation.py
def queenPermutation(board, tq, qpsf,ans):
    if tq == qpsf:
        print(ans)
        return
    
    for i in range(len(board)):
        if board[i] == False:
            board[i] = True
            queenPermutation(board, tq, qpsf + 1,ans + "q" + str(qpsf) + "b" + str(i) + "  ")
            board[i] = False

board = [False] * 4
tq = 2
qpsf = 0
ans = ""
queenPermutation(board, tq, qpsf,ans)

File: final_prog/backtracking/2_1DQueenpermutation.java
import java.util.*;

class Queen1D{
    public static void main(String[] args) {
        int tq = 2; // total no. of Queens

        int qpsf = 0; // queen placed so far

        String ans = ""; // answer

        boolean[] board = new boolean[4]; // true -> queen is placed i that box, false -> box is empty

        queenPermutation(board, tq, qpsf, ans);
    }

    public static void queenPermutation(boolean[] board,int tq,int qpsf,String ans){
        if(tq == qpsf){
            System.out.println(ans);
            return;
        }

        for(int i = 0;i < board.length;i++){
            if(board[i] == false){
                board[i] = true; // place queen at box[i].
                queenPermutation(board, tq, qpsf + 1, ans + "q" + qpsf + "b" + i + " ");
                board[i] = false; // unplace queen at box[i] or UNDO step
            }
        }
    }


}